### PR TITLE
Remove nstar codec definition

### PR DESCRIPTION
**Pull request by alfonsoros88**
This changes intends to simplify a bit `nstar`'s  source by removing the
codec definition to serialize messages.

### CODE REVIEW

File: nstar/src/main.rs [hunk 1]
```diff
@@ -107,8 +108,9 @@ async fn main() -> Result<()> {
         };
         info!("Connected to {}", opt.host);
         let (read, write) = stream.into_split();
-        let mut framed_read = tokio_util::codec::FramedRead::new(read, codec::Codec::default());
-        let mut framed_write = tokio_util::codec::FramedWrite::new(write, codec::Codec::default());
+        let mut framed_read = tokio_util::codec::FramedRead::new(read, LengthDelimitedCodec::new());
```
Comment (flxo) [inline nstar/src/main.rs:111]: Map the stream to `api::Message` to have the deserialisation at only one place.

**Review by flxo — CHANGES_REQUESTED** [review]

File: nstar/src/main.rs [hunk 2]
```diff
@@ -272,7 +284,7 @@ async fn main() -> Result<()> {
                                     while let Some(r) = npk.next().await {
                                         match r {
                                             // Send the chunk to the stream
-                                            Ok(b) => match framed_write.send(codec::Message::Raw(b)).await {
+                                            Ok(b) => match framed_write.send(b).await {
```
[the remark this answers is not in the document]
Comment (alfonsoros88) [reply nstar/src/main.rs:287]: whoops, I overlooked that. Will close this pr.

**Comment by flxo** [comment]
Think we can fix that :-)

**Comment by marcmo** [comment]
@flxo do we still want to pursue this PR?

**Comment by flxo** [comment]
> @flxo do we still want to pursue this PR?

I have some more things for the codec in the pipeline for the quiets christmas days. I like the codec idea because it can be reused but the TCP stream part is tricky. Let's close this here.

**Pull request closed without merging**